lib/iris/fileformats/rules.py

```diff
@@ -735,7 +735,8 @@ def _ensure_aligned(regrid_cache, src_cube, target_cube):
 
 Loader = collections.namedtuple('Loader',
                                 ('field_generator', 'field_generator_kwargs',
-                                 'converter', 'legacy_custom_rules'))
+                                 'converter', 'legacy_custom_rules',
+                                 'fault_tolerant'))
```

Comment:
The alternative is that we just make the GRIB loader fault_tolerant globally. Provided the warning type is specific to GRIB loading, this can even be a case of:

```
warn(GRIBLoadFault, 'A specific bad thing happened')
```

And if we want to change the tolerance, we can change what happens when a GRIBLoadFault warning is encountered (e.g. make it raise an exception) through the standard warning mechanisms.